client/state/data-layer/wpcom/sites/posts/likes/mine/delete/index.js

```diff
@@ -0,0 +1,51 @@
+/** @format */
+
+/**
+ * External Dependencies
+ */
+
+/**
+ * Internal Dependencies
+ */
+import { like, removeLiker } from 'state/posts/likes/actions';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { POST_UNLIKE } from 'state/action-types';
+import { bypassDataLayer } from 'state/data-layer/utils';
+
+export function fromApi( response ) {
+	if ( ! response.success ) {
+		throw new Error( 'Unsuccessful unlike API request' );
+	}
+	return {
+		likeCount: +response.like_count,
+		liker: response.liker,
+	};
+}
+
+export const fetch = action =>
+	http(
+		{
+			method: 'POST',
+			path: `/sites/${ action.siteId }/posts/${ action.postId }/likes/mine/delete`,
+			apiVersion: '1.1',
+			body: {},
+		},
+		action
+	);
+
+export const onSuccess = ( { siteId, postId }, { likeCount, liker } ) =>
+	removeLiker( siteId, postId, likeCount, liker );
+
+export const onError = ( { siteId, postId } ) => bypassDataLayer( like( siteId, postId ) );
+
+export default {
+	[ POST_UNLIKE ]: [
```

Comment:
I like to encourage keeping with the difference between requests and fact even with `POST`s

We may want to like a post but it may not be possible. We also end up with weird things like `POST_LIKES_REMOVE_LIKER` when `POST_UNLIKE` is probably what we'd rather dispatch in order to have the reducers remove the like.

```js
[ POST_UNLIKE_REQUEST ]: [ dispatchRequestEx( {
  fetch,
  onSuccess: ( { siteId, postId }, { likeCount, liker } ) => [
    { type: POST_UNLIKE, siteId, postId, liker },
    { type: POST_LIKES_SET, siteId, postId, likeCount }
  ]
} ]
```

something like that